File: DSA/Python/UseSideLane.py
# Use Side Lane

def solve(n, arr):
  stack = []
  curr = 1
  for i in range(n):
    truck = arr[i]
    while stack and stack[-1] == curr:
      stack.pop()
      curr += 1
    if truck == curr:
      curr += 1
    else:
      stack.append(truck)
  while stack and stack[-1] == curr:
    stack.pop()
    curr += 1
  if not stack:
    print("yes")
  else:
    print("no")


# OR 

# With Input 

def solve(n, arr):
  stack = []
  curr = 1
  for i in range(n):
    truck = arr[i]
    while stack and stack[-1] == curr:
      stack.pop()
      curr += 1
    if truck == curr:
      curr += 1
    else:
      stack.append(truck)
  while stack and stack[-1] == curr:
    stack.pop()
    curr += 1
  if not stack:
    print("yes")
  else:
    print("no")
    
while True:
  line = input().strip()
  if line == "0":
    break
  n = int(line)
  arr = list(map(int, input().strip().split()))
  solve(n, arr)

File: DSA/JS/DSA/UseSideLane.js
// Use Side Lane 


function solve(n, arr) {
  let st = []
  let curr = 1
  for (let i = 0; i < n; i++) {
    let tr = arr[i]
    while (st.length > 0 && st[st.length - 1] === curr) {
      st.pop()
      curr++
    }
    if (tr === curr) {
      curr++
    } else {
      st.push(tr)
    }
  }
  while (st.length > 0 && st[st.length - 1] === curr) {
    st.pop()
    curr++
  }
  if (st.length === 0) {
    console.log("yes")
  } else {
    console.log("no")
  }
}
